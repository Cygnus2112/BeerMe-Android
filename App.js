import React from 'react';
import { Provider } from 'react-redux';
import 'react-native-gesture-handler';
import { NavigationContainer } from '@react-navigation/native';
import { createStackNavigator } from '@react-navigation/stack';

const Stack = createStackNavigator();

import store from './reducers/rootStore';

import Login from './views/login';
import Signup from './views/signup';
import Styles from './views/styles';
import Swipe from './views/swipe';
import Main from './views/main';
import Wishlist from './views/wishlist';
import BeerDetail from './views/beerdetail';
import Browser from './views/webview';
import Forgot from './views/forgot';
import Whatever from './views/whatever';
import About from './views/about';

// BackAndroid.addEventListener('hardwareBackPress', () => {
//     Actions.pop();
//     return true;
// });

const App: () => React$Node = () => {
  return (
    <>
      <Provider store={store}>
        <NavigationContainer>
          <Stack.Navigator headerMode={'none'}>
            <Stack.Screen
              name="main"
              component={Main}
              options={{ animationEnabled: false }}
            />
            <Stack.Screen
              name="login"
              component={Login}
              options={{ animationEnabled: false }}
            />
            <Stack.Screen
              name="signup"
              component={Signup}
              options={{ animationEnabled: false }}
            />
            <Stack.Screen
              name="styles"
              component={Styles}
              options={{ animationEnabled: false }}
            />
            <Stack.Screen
              name="swipe"
              component={Swipe}
              options={{ animationEnabled: false }}
            />
            <Stack.Screen
              name="wishlist"
              component={Wishlist}
              options={{ animationEnabled: false }}
            />
            <Stack.Screen
              name="beerdetail"
              component={BeerDetail}
              options={{ animationEnabled: false }}
            />
            <Stack.Screen
              name="webview"
              component={Browser}
              options={{ animationEnabled: false }}
            />
            <Stack.Screen
              name="forgot"
              component={Forgot}
              options={{ animationEnabled: false }}
            />
            <Stack.Screen
              name="whatever"
              component={Whatever}
              options={{ animationEnabled: false }}
            />
            <Stack.Screen
              name="about"
              component={About}
              options={{ animationEnabled: false }}
            />
          </Stack.Navigator>
        </NavigationContainer>
      </Provider>
    </>
  );
};

export default App;
