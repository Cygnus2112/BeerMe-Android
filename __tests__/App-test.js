/**
 * @format
 */

import 'react-native';
import React from 'react';
import App from '../App';
import { Provider } from 'react-redux';
import renderer, { act } from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import configureStore from 'redux-mock-store';

import Wishlist from '../views/wishlist';

jest.useFakeTimers();
jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper');

configure({ adapter: new Adapter() });

const getMockProvider = (state) => {
  const mockStore = configureStore([]);
  const store = mockStore(state);
  return {
    MockProvider: ({ children }) => (
      <Provider store={store}>{children}</Provider>
    ),
    store,
  };
};

describe('Testing the App', () => {
  it('renders without crashing', async () => {
    const rendered = renderer.create(<App />).toJSON();
    await act(async () => {});
    expect(rendered).toBeTruthy();
  });

  it('renders correctly', async () => {
    const tree = renderer.create(<App />).toJSON();
    await act(async () => {});
    expect(tree).toMatchSnapshot();
  });
});

describe('Testing Wishlist', () => {
  jest.mock('react-redux', () => ({
    useSelector: jest.fn(),
    useDispatch: () => jest.fn(),
  }));
  const wishlist = [
    {
      "abv": "5.8",
      "brewery": "SweetWater Brewing Company", "descript": "An American Brown Ale, deep copper-colored, mild malt.",
      "icon": "https://brewerydb-images.s3.amazonaws.com/beer/9cqNKc/upload_fyCyII-icon.png",
      "id": "9cqNKc",
      "label": "https://brewerydb-images.s3.amazonaws.com/beer/9cqNKc/upload_fyCyII-medium.png",
      "name": "Georgia Brown", 
      "style": "American-Style Brown Ale", "website": "http://www.sweetwaterbrew.com/"
    },
    {
      "abv": "8.5",
      "brewery": "Laughing Dog Brewing",
      "descript": "A True Hop Bomb Brewed plenty of Columbus and Mt Hood Hops for a Piney hop character.",
      "icon": "https://brewerydb-images.s3.amazonaws.com/beer/aG4Ie2/upload_yX4wkY-icon.png",
      "id": "aG4Ie2",
      "label": "https://brewerydb-images.s3.amazonaws.com/beer/aG4Ie2/upload_yX4wkY-medium.png",
      "name": "Alpha Dog Imperial IPA",
      "style": "Imperial or Double India Pale Ale",
      "website": "http://www.laughingdogbrewing.com/"
    }, {
      "abv": "8.2",
      "brewery": "Lagunitas Brewing Company",
      "descript": "Caution must be exercised when drinking this beer.",
      "icon": "https://brewerydb-images.s3.amazonaws.com/beer/fnqCGB/upload_IaFcNP-icon.png",
      "id": "fnqCGB",
      "label": "https://brewerydb-images.s3.amazonaws.com/beer/fnqCGB/upload_IaFcNP-medium.png",
      "name": "Maximus",
      "style": "American-Style India Pale Ale", "website": "http://www.lagunitas.com/"
    }
  ];

  it('renders as expected', async () => {
    const state = {
      wishlistReducer: {
        wishlist,
      }
    };

    const { MockProvider } = getMockProvider(state);

    const wrapper = shallow(
      <MockProvider>
        <Wishlist />
      </MockProvider>
    );

    expect(wrapper.dive()).toMatchSnapshot();
  });
});

