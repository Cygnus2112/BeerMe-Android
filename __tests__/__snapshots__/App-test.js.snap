// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Wishlist renders as expected 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <Wishlist />
</ContextProvider>
`;

exports[`Testing the App renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onOpen={[Function]}
          onTransitionStart={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              collapsable={false}
              forwardedRef={[Function]}
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "overflow": "hidden",
                    },
                    Array [
                      Object {
                        "backgroundColor": "rgb(242, 242, 242)",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "flexDirection": "column-reverse",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "backgroundColor": "#ddd",
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "#ddd",
                            "flex": 1,
                            "justifyContent": "center",
                          }
                        }
                      >
                        <Image
                          source={
                            Object {
                              "testUri": "../../../assets/logo_outline.png",
                            }
                          }
                        />
                      </View>
                      <View
                        style={
                          Object {
                            "flex": 0.2,
                          }
                        }
                      />
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaProvider>
</View>
`;
